core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2AuthenticationSupportModule.java

```diff
@@ -26,6 +26,7 @@
     protected void setup(Binder binder)
```

Comment:
> Fix OIDC implementation to resolve correctness issues

Can you please explain what was the issue about?

Reply:
Changed the commit message to be clearer. Specifically the commit fixes several structural issues where there were inappropriate dependencies and duplicate/confusing code. Additionally there are several security fixes including correctly validating ID and access tokens.

Reply:
I worked with Nik on this change and agreed to the current structure.